src/components/ResourceMachine.tsx

```diff
@@ -1,18 +1,70 @@
 import React from 'react';
-import {Text, View, TouchableOpacity} from 'react-native';
+import {StyleSheet} from 'react-native';
+import {Machine, ResourceType} from '../types';
+import {useResource, useResourceCount} from '../hooks';
+import ThemedButton from './ThemedButton';
+import ThemedView from './ThemedView';
+import ThemedText from './ThemedText';
+import ResourceCost from './ResourceBullet';
 
-const ResourceMachine = () => {
+const styles = StyleSheet.create({
+  machineCard: {
+    alignSelf: 'center',
+    flexDirection: 'column',
+    flexWrap: 'wrap',
+    alignItems: 'flex-start',
+  },
+  valueContainer: {
+    flexDirection: 'row',
+    flexWrap: 'wrap',
+    alignItems: 'flex-start',
+  },
+  valueBox: {
+    width: '50%',
+  },
+});
+
+const renderCost = (resourceType: ResourceType, cost: number | undefined) => {
+  const data = useResource(resourceType);
+  const count = useResourceCount(resourceType);
+
+  return (
+    <ResourceCost
+      name={data.name}
+      dps={count.perSecond}
+      cost={cost ? cost : 2}
+      current={count.current}
+      storage={count.capacity}
+    />
+  );
+};
+
+const ResourceMachine = ({machine}: {machine: Machine}) => {
   return (
-    <View>
-      <Text>Miner: 2</Text>
-      <Text>Build a pickaxe for your miner.</Text>
-      <Text>Cost:</Text>
-      <Text>Input:</Text>
-      <Text>Output:</Text>
-      <TouchableOpacity>
-        <Text>Get 1</Text>
-      </TouchableOpacity>
-    </View>
+    <ThemedView style={styles.machineCard}>
+      <ThemedView>
+        <ThemedText variant={'heading'}>{machine.name}</ThemedText>
```

Comment:
When it's a static text you can omit the `{}`. So you can replace:

`<ThemedText variant={'heading'}>`
with
`<ThemedText variant='heading'>`